src/input_handlers.h

```diff
@@ -78,23 +78,76 @@ class RecvFromInputHandler : public MessageParser<InPlaceAccumulation> {
+                        if (result == DOCA_ERROR_NO_MEMORY) {
+                            nanosleep(&ts, &ts);
+                        }
+                    } while (s_user_params.is_blocked && result == DOCA_ERROR_NO_MEMORY);
+                    if (result != DOCA_SUCCESS) {
+                        if (result == DOCA_ERROR_NO_MEMORY) { // only for non-blocked
+                            errno = EAGAIN;
+                            ctx->buf_size = -1;
+                        } else {
+                            log_err("Doca doca_cc_consumer_post_recv_task_alloc_init failed");
+                            ret = RET_SOCKET_SHUTDOWN;
+                        }
+                    } else {
+                        do {
+                            result = doca_cc_consumer_post_recv_task_alloc_init(ctx->ctx_fifo.consumer, doca_buf, &consumer_task);
+                            if (result == DOCA_ERROR_NO_MEMORY) {
+                                nanosleep(&ts, &ts);
+                            }
+                        } while (s_user_params.is_blocked && result == DOCA_ERROR_NO_MEMORY);
+                        if (result != DOCA_SUCCESS) {
+                            if (result == DOCA_ERROR_NO_MEMORY) { // only for non-blocked
+                                errno = EAGAIN;
+                                ctx->buf_size = -1;
+                                (void)doca_buf_dec_refcount(doca_buf, NULL);
+                            } else {
+                                log_err("Doca doca_cc_consumer_post_recv_task_alloc_init failed");
+                                ret = RET_SOCKET_SHUTDOWN;
+                            }
+                        } else {
+                            task_obj = doca_cc_consumer_post_recv_task_as_task(consumer_task);
+                            result = doca_task_submit(task_obj);
+                            if (result != DOCA_SUCCESS) {
+                                log_err("failed submitting recv task with error = %s", doca_error_get_name(result));
+                                doca_task_free(task_obj);
+                                (void)doca_buf_dec_refcount(doca_buf, NULL);
+                                return RET_SOCKET_SHUTDOWN;
+                            }
+                            if (s_user_params.mode == MODE_CLIENT && !g_pApp->m_const_params.b_client_ping_pong
+                                && !g_pApp->m_const_params.b_stream) { // latency_under_load
+                                pe_local = s_user_params.pe_underload;
+                            }
+                        }
+                    }
+                }  // end of doca fast path
+                // Waiting for meesage receive callback - blocking or fifo mode and task submitted successfully
+                if (s_user_params.is_blocked || (s_user_params.doca_fast_path && result == DOCA_SUCCESS)) {
```

Comment:
submit recv
wait for recv complete
pending

need to save flag if we got eagain after task submit